ql/src/java/org/apache/hadoop/hive/ql/security/authorization/command/CommandAuthorizerV2.java

```diff
@@ -185,6 +191,33 @@ private static void addHivePrivObject(Entity privObject, Map<String, List<String
           tableName2Cols.get(Table.getCompleteName(table.getDbName(), table.getTableName()));
       hivePrivObject = new HivePrivilegeObject(privObjType, table.getDbName(), table.getTableName(),
           null, columns, actionType, null, null, table.getOwner(), table.getOwnerType());
+      if(table.getStorageHandler() != null){
+        //TODO: add hive privilege object for storage based handlers for create and alter table commands.
+        if(hiveOpType == HiveOperationType.CREATETABLE ||
+                hiveOpType == HiveOperationType.ALTERTABLE_PROPERTIES ||
+                hiveOpType == HiveOperationType.CREATETABLE_AS_SELECT){
+          String storageuri = null;
+          Map<String, String> tableProperties = new HashMap<>();
+          Configuration conf = new Configuration();
+          tableProperties.putAll(table.getSd().getSerdeInfo().getParameters());
+          tableProperties.putAll(table.getParameters());
+          try {
+            if(table.getStorageHandler() instanceof HiveStorageAuthorizationHandler){
+              HiveStorageAuthorizationHandler authorizationHandler = (HiveStorageAuthorizationHandler) ReflectionUtils.newInstance(
+                      conf.getClassByName(table.getStorageHandler().getClass().getName()), SessionState.get().getConf());
+              storageuri = authorizationHandler.getURIForAuth(tableProperties).toString();
+            }else{
+              //Custom storage handler that has not implemented the HiveStorageAuthorizationHandler
+              storageuri = table.getStorageHandler().getClass().getName()+"://"+
+                      HiveCustomStorageHandlerUtils.getTablePropsForCustomStorageHandler(tableProperties);
+            }
+          }catch(Exception ex){
+            ex.printStackTrace();
```

Comment:
can you either log this exception to the log file if this is concerning to the user or ignore it if it not an issue, with a comment instead of printStackTrace() ? This goes to STDOUT and not to the server log.